### PR TITLE
Core Concepts week 1 day 2 updated

### PR DESCRIPTION
**Pull request by beth-panx**
---

⚠️ Merge/Release date 2/15

---

Updates:
- Week 1 day 2 article

***Please ignore day 1 changes. They are included in this PR #27**

### CODE REVIEW

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 1]
```diff
@@ -1,16 +1,174 @@
...
+
+> Want to audit your PWA for security? Look at the _Security_ tab on the PWABuilder report. Here's what that looks like for the sample PWA.
+
+![DevTools Tips - Report Card](_media/dtt-security.png)
+
+Getting a good Security score is the simplest requirement to meet. Just **publish your PWA to an HTTPS-enabled endpoint**. You have two options:
+ * Use a hosting service that supports HTTPS by default. Cloud Providers like [Azure](https://aka.ms/learn-PWA/30Days-1.2/azure.microsoft.com) offer options like [Azure App Service](https://aka.ms/learn-PWA/30Days-1.2/azure.microsoft.com/services/app-service/web) that can help. 
+ * Use your own hosting options, and create the required certificates using free services like [Lets Encrypt](https://aka.ms/learn-PWA/30Days-1.2/letsencrypt.org/docs).
+
+Note that Microsoft Edge will let you use `http://localhost` (non-HTTPS) to preview and test your PWA locally - _for debugging purposes only_.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:78]:
```suggestion
Note that browsers will let you use `http://localhost` (non-HTTPS) to preview and test your PWA locally - _for debugging purposes only_.
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 2]
```diff
@@ -1,16 +1,174 @@
...
+
+---
+
+## 2. Service Workers
+
+[Service Workers](https://aka.ms/learn-PWA/30Days-1.2/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/service-workers) are a special type of [Web Worker](https://aka.ms/learn-PWA/30Days-1.2/www.w3schools.com/js/js_api_web_workers.asp), a JavaScript task that can run in the background (for asynchronous or long-running operations) without impacting the performance of the page. 
+
+Service Workers make your PWA _reliable_ and _network-independent_, ensuring that it provides a usable experience under flaky network conditions or when device is offline (ex: _Flight Mode_). It achieves this with two key capabilities:
+ * **Fetch API**. Service Workers can intercept, modify and respond to all network requests from the application dynamically. They can listen for events indicating network changes, adapting their fetch responses based on the runtime conditions and app context.
+ * **Cache API**. Service Workers can access client-side cache and asynchronous storage options (like IndexedDB, Local & Session Storage) to proactively store resources for offline access or performance efficiency.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:88]:
```suggestion
 * **Cache API**. Service Workers can access client-side Cache as well as asynchronous storage options (like IndexedDB) to proactively store resources for offline access or performance efficiency.
```

Comment (aarongustafson) [reply docs/30DaysOfPWA/core-concepts/02.md:88]: `localStorage` & `sessionStorage` are synchronous APIs and unavailable to Service Workers.

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 3]
```diff
@@ -1,16 +1,174 @@
...
+ * Click an artcle. What happens? _You may get the "You Are Offline" message if you have not visited it before_. 
+ * Uncheck the _Offline_ button. Revisit the article (it loads) then check _Offline_ again. _You should be able to move between Home and Article pages and see them both offline_.
+
+That's the power of Fetch (interception) and Cache (storage) delivering a native-like experience (works offline) for the PWA!
+
+> Want to audit your PWA for Service Worker support? Look for the tab in the PWA Builder report. Here's what that looks like for DevTools Tips. 
+
+![DevTools Tips - Report Card](_media/dtt-sw.png)
+
+We see it meets basic requirements (_has a service worker_) and implements some recommended and optional behaviors (_works offline_, _uses periodic sync_, _uses background sync_) for a better PWA experience.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:104]:
```suggestion
We see it meets the basic requirements (_has a service worker_) and implements some recommended and optional behaviors (_works offline_, _uses periodic sync_, _uses background sync_) for a better PWA experience.
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 4]
```diff
@@ -1,16 +1,174 @@
...
+
+We'll talk about Service Worker operations (scope, lifecycle) and caching strategies (for offline and performance needs) in a later post. 
+
+---
+
+## 3. Web App Manifest
+
+So we've made the PWA secure and ensure reliable, network-independent operation. But how do we make it _installable_ like a native app?
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:112]:
```suggestion
So we've made the PWA secure and ensured reliable, network-independent operation. But how do we make it _installable_ like a native app? 
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 5]
```diff
@@ -1,16 +1,174 @@
...
+
+We'll talk about Service Worker operations (scope, lifecycle) and caching strategies (for offline and performance needs) in a later post. 
+
+---
+
+## 3. Web App Manifest
+
+So we've made the PWA secure and ensure reliable, network-independent operation. But how do we make it _installable_ like a native app? 
+
+That's where the [Web Application Manifest](https://aka.ms/learn-PWA/30Days-1.2/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/web-app-manifests) comes in. It's a [W3C Specfication](https://aka.ms/learn-PWA/30Days-1.2/www.w3.org/TR/appmanifest) helps. It's a specification that defines **a JSON-based file format** that provides developers with a **centralized place to put metadata** associated with apps.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:114]:
```suggestion
That's where the [Web Application Manifest](https://aka.ms/learn-PWA/30Days-1.2/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/web-app-manifests) comes in. It's a [W3C Specification](https://aka.ms/learn-PWA/30Days-1.2/www.w3.org/TR/appmanifest) defining **a JSON-based file format** that provides developers with a **centralized place to put metadata** associated with apps.
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 6]
```diff
@@ -1,16 +1,174 @@
...
+
+From a PWA developer perspective, it's a 2-step process:
+ * Create your _manifest.json_ file and associate it with your PWA by referencing it in the `<head>` section of your HTML page (see below).
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:117]:
```suggestion
 * Create your _manifest.json_ file and associate it with your PWA by referencing it in the `head` of your HTML page (see below). 
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 7]
```diff
@@ -1,16 +1,174 @@
...
+
+```html
+<link rel="manifest" href="/manifest.json">
+```
+
+> Let's see this in action with our sample PWA
+
+Go back to the _Inspect_ tab we opened in Task 1.
+ * Scan the `Elements` tab - do you see the link in `<head>`?
+ * Scan the `Applications` tab - do you see the `manifest.json` file?
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:128]: They could also see when it’s loaded in the Network tab :-)

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 8]
```diff
@@ -1,16 +1,174 @@
...
+
+Here's what that looks like for DevTools Tips. Note that the dashboard also shows you currently populated Manifest properties grouped into categories (Identity, Presentation, Icons etc.) for convenience. We'll dive into the details in a future post. 
+
+![DevTools Tips - Report Card](_media/dtt-manifest.png)
+
+
+> Want to audit your PWA for Manifest support? 
+
+Look at your PWABuilder report from Task 2. Here's what that looks like for DevTools Tips.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:137]:
```suggestion
Look at your PWABuilder report from Task 2. Here's what that looks like for DevTools Tips:
```

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 9]
```diff
@@ -1,16 +1,174 @@
...
+> How does Web App Manifest support Installability?
+
+It governs how the PWA looks and behaves when installed on your target device platform. It is used by the browser to determine that a website is an installable PWA - and to provide the relevant [install experience](https://aka.ms/learn-PWA/30Days-1.2/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/ux#installing-a-pwa) to users. It also supports direct installs from app stores when you [publish your PWA](https://aka.ms/learn-PWA/30Days-1.2/docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/microsoft-store).
+
+## Summary
+
+![Summary Day 2 - Deconstructing PWA.](_media/deconstruct-pwa.png)
+
+That was a lot to cover, but here are the core takeaways:
+ - PWAs use open web technologies to implement progressive-enhancement techniqiues in a cross-brower compatible way.
```
Comment (aarongustafson) [inline docs/30DaysOfPWA/core-concepts/02.md:152]:
```suggestion
 - PWAs use open web technologies to implement progressive-enhancement techniques in a cross-brower compatible way.
```

**Review by aarongustafson — CHANGES_REQUESTED** [review]
Minor edits

File: docs/30DaysOfPWA/core-concepts/02.md [hunk 10]
```diff
@@ -1,16 +1,174 @@
...
+
+```html
+<link rel="manifest" href="/manifest.json">
+```
+
+> Let's see this in action with our sample PWA
+
+Go back to the _Inspect_ tab we opened in Task 1.
+ * Scan the `Elements` tab - do you see the link in `<head>`?
+ * Scan the `Applications` tab - do you see the `manifest.json` file?
```
Comment (beth-panx) [inline docs/30DaysOfPWA/core-concepts/02.md:128]:
```suggestion
 * Scan the `Applications` or `Network` tab - do you see the `manifest.json` file?
```

**Comment by beth-panx** [comment]
@aarongustafson - I updated the article based on your suggestions. Only one about HTTPS on GitHub page that I didn't change.

**Review by zateutsch — APPROVED** [review]

**Pull request merged**